delegate/shared/src/main/scala/zio/macros/delegate/package.scala

```diff
@@ -24,4 +26,44 @@ package object delegate {
   def patch[A, B](implicit ev: A Mix B): (B => B) => A with B => A with B =
     f => old => ev.mix(old, f(old))
 
+  /**
+   * Create a function that can be used to enrich ZIO or ZManaged with the mixed in trait instance.
+   */
+  def enrichWith[A](a: A): EnrichWith[A] = new EnrichWith(a)
+
+  /**
+   * Create a function that can be used to enrich ZIO or ZManaged with the mixed in effectfully created trait instance.
+   * The generating effect can use the output of the effect that will be mixed into.
+   */
+  def enrichWithM[A]: EnrichWithM.PartiallyApplied[A] = new EnrichWithM.PartiallyApplied
+
+  /**
+   * Create a function that can be used to enrich ZIO or ZManaged with the mixed in effectfully created trait instance.
+   * The generating effect can use the output of the effect that will be mixed into.
+   */
+  def enrichWithManaged[A]: EnrichWithManaged.PartiallyApplied[A] = new EnrichWithManaged.PartiallyApplied
+
+  implicit class ZIOSyntax[R, E, A](zio: ZIO[R, E, A]) {
+
+    def >>+[B](enrichWith: EnrichWith[B])(implicit ev: A Mix B): ZIO[R, E, A with B] =
```

Comment:
I like the symbolic alias, but question to @jdegoes: is `>>+` free to take? It's not currently used in ZIO, but perhaps it already has estabilished meaning in some context I'm not aware of. Just to double check.